### DISCUSSION TITLE
Adding a "CI" mode to the app?

### DISCUSSION THREAD

**Discussion by Lord-Kamina** [performous/performous#1010] [Ideas]
This is an idea I've been toying around in my head for a while, and would like opinions from @performous/contributors  and especially @performous/core-owners.

Our CI is IMO pretty decent by now., but it only tells us if the code,
 a) Builds
 b) Behaves as expected under controlled conditions.

But, often that is not enough. I remember a while ago I noticed the mingw builds didn't actually run. They built, tests passed but I tried to download and run them on several different windows computers, and they just wouldn't run (in fact, I'm not entirely sure if that was ever fixed)

So how about this? Maybe we should add a "CI" mode implemented as follows: Add a new command-line option that sets a switch that is checked during the `mainLoop` (we could disable it for Release mode maybe, if it turned out to compromise performance, although I doubt it).

If this switch is on, performous behaves as if the quit option had been selected and tried to exit gracefully.

Then, we could run this mode on the CI and have an idea whether Performous is actually able to run and exit normally on that platform.

What do you think?

**Comment by Baklap4** [comment]
Sounds like a great idea!
